tests/server/routers/test_structures.py

```diff
@@ -154,3 +154,32 @@ def test_structures_endpoint_data(self):
         assert len(self.json_response["data"]) == 2
         assert "included" in self.json_response
         assert len(self.json_response["included"]) == 2
+
+
+class TestStructuresWithNullFieldsDoNotMatchNegatedFilters(RegularEndpointTests):
+    """Tests that structures with e.g., `'assemblies':null` do not get
+    returned for negated queries like `filter=assemblies != 1`, as mandated
+    by the specification.
+
+    """
+
+    request_str = "/structures?filter=assemblies != 1"
```

Comment:
(I assume all these tests run on the same test dataset.)
Would it not be better to test this with a different field descriptor than 'assemblies'?
'assemblies' is one of the fields defined in the current Optimade standard. It may very well happen that in the future we want to have real values for the assemblies field in our test data. In that case this test would fail as more than 0 documents would be returned.

Reply:
We could, yes, but obviously as we control the test data, this is the easiest way to proceed for now. Assemblies is not really intended to be a queryable field anyway as it contains some nested objects, so I think this is fairly safe.